### ISSUE TITLE
No way to know if Transitions to same Scene class is backwards

### ISSUE THREAD

**Issue by gumil** [nhaarman/acorn#151]
**Is your feature request related to a problem? Please describe.**
For example:
Scene1 pushes to Scene1
They are the same scenes but can have different data. So during transition there's no way to identify if the scene is going back one screen since it is still the same scene class.

An example of a transition that affects this is sliding from left to right. When going back it should be from right to left.

**Describe the solution you'd like**
Have a way in transition factory to know if the transition happening is going back or forward.

**Describe alternatives you've considered**
Is it also possible for a more granular transition mechanism where it can be specified on specific navigators?

This is an edge case it seems as I am trying this out in a sample application with unrealistic backstack and screens.

**Comment by nhaarman** [comment]
There is [`TransitionData`](https://github.com/nhaarman/acorn/blob/1.x/core/acorn-core/src/main/java/com/nhaarman/acorn/navigation/TransitionData.kt#L31) which contains an `isBackwards` property, and is passed to the `SceneTransitionFactory`. 

For the `StackNavigator` this data is properly set when pushing and popping and is [used for the default transitions](https://github.com/nhaarman/acorn/blob/1.x/ext/acorn-android/src/main/java/com/nhaarman/acorn/android/transition/DefaultSceneTransitionFactory.kt#L36) accordingly.

Is this what you're looking for?

**Comment by gumil** [comment]
Yes this is it. Thanks for the help. This feature is kinda hard to find. It was not in the documentation and I assumed acorn creates it using only the DSL

**Comment by nhaarman** [comment]
Cool! I'll make sure it gets added to the docs :+1:
